src/lib/message/L1ToL2MessageCreator.ts

```diff
@@ -25,60 +62,85 @@ export class L1ToL2MessageCreator {
-      gasEstimationOptions?: GasOverrides
-    },
-    overrides: PayableOverrides = {}
-  ): Promise<L1TransactionReceipt> {
+  public async getTicketCreationRequest(
+    params: L1ToL2MessageParams
+  ): Promise<L1ToL2TransactionRequest> {
     const sender = await this.l1Signer.getAddress()
-    const excessFeeRefundAddress = options?.excessFeeRefundAddress || sender
-    const callValueRefundAddress = options?.callValueRefundAddress || sender
+    const excessFeeRefundAddress =
+      params.options?.excessFeeRefundAddress || sender
+    const callValueRefundAddress =
+      params.options?.callValueRefundAddress || sender
     const l1Provider = SignerProviderUtils.getProviderOrThrow(this.l1Signer)
 
-    const gasEstimator = new L1ToL2MessageGasEstimator(l2Provider)
+    const gasEstimator = new L1ToL2MessageGasEstimator(params.l2Provider)
     const baseFee = await getBaseFee(l1Provider)
-    const defaultedGasParams = await gasEstimator.estimateAll(
+    const estimates = await gasEstimator.estimateAll(
       sender,
-      l2CallTo,
-      l2CallData,
-      l2CallValue,
+      params.l2CallTo,
+      params.l2CallData,
+      params.l2CallValue,
       baseFee,
       excessFeeRefundAddress,
       callValueRefundAddress,
       l1Provider,
-      options?.gasEstimationOptions
+      params.options?.gasEstimationOptions
     )
 
-    const l2Network = await getL2Network(l2Provider)
-    const inbox = Inbox__factory.connect(
-      l2Network.ethBridge.inbox,
-      this.l1Signer
+    const l2Network = await getL2Network(params.l2Provider)
+    const inboxInterface = Inbox__factory.createInterface()
+    const functionData = inboxInterface.encodeFunctionData(
+      'createRetryableTicket',
+      [
+        params.l2CallTo,
+        params.l2CallValue,
+        estimates.maxSubmissionFee,
+        excessFeeRefundAddress,
+        callValueRefundAddress,
+        estimates.gasLimit,
+        estimates.maxFeePerGas,
+        params.l2CallData,
+      ]
     )
 
-    const res = await inbox.createRetryableTicket(
-      l2CallTo,
-      l2CallValue,
-      defaultedGasParams.maxSubmissionFee,
-      excessFeeRefundAddress,
-      callValueRefundAddress,
-      defaultedGasParams.gasLimit,
-      defaultedGasParams.maxFeePerGas,
-      l2CallData,
-      {
-        value: defaultedGasParams.totalL2GasCosts.add(l2CallValue),
-        ...overrides,
-      }
-    )
-    const receipt = await res.wait()
+    return {
+      l2GasLimit: estimates.gasLimit,
+      l2MaxFeePerGas: estimates.maxFeePerGas,
+      l2SubmissionFee: estimates.maxSubmissionFee,
+      l2GasCostsMaxTotal: estimates.totalL2GasCosts,
+      txRequestCore: {
+        to: l2Network.ethBridge.inbox,
+        data: functionData,
+        value: estimates.totalL2GasCosts.add(params.l2CallValue),
+      },
+    }
+  }
+
+  /**
+   * Creates a retryable ticket by directly calling the Inbox contract on L1
+   */
+  public async createRetryableTicket(
```

Comment:
I like this approach of having a separate function to get the request, another option we could take is have an override optional parameter which makes this same function return the request instead of submitting the tx
I guess the tradeoff is less discoverability when browsing the api, but makes it neater to toggle any tx into a request with minimal changes on the consumer side
(I don't know if it would necessarily be better, but bringing it up for discussion)

Reply:
yeh, I think I like the discoverability. It will also make for more readable code for the client.